app/components/publications/publication/publication-navigation.js

```diff
@@ -0,0 +1,18 @@
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { task } from 'ember-concurrency-decorators';
+
+export default class PublicationNavigation extends Component {
+  @tracked documentsCount
+
+  constructor() {
+    super(...arguments);
+    this.loadData.perform();
+  }
+
+  @task
+  *loadData() {
+    const pieces = yield this.args.publicationFlow.case.get('pieces');
```

Comment:
2-ledige opmerking:
- er is een catch aan het gebruiken van de relatie-promise (`.pieces`/`get('pieces')`). Deze zal in de achtergrond een JSONAPI request van de vorm `[GET] case/123/pieces` tot gevolg hebben. Gezien deze request geen `page[size]` expliciteert, zal de default page size (meestal `20`) gebruikt worden. Hierdoor krijg je bij  hoge aantallen niet álle, maar enkel *#page_size*-objecten terug. Voor plaatsen waar je met zekerheid weet dat er maar een klein aantal elementen zullen terugkomen, is dit dus betrouwbaar, maar niet voor de documenten van een dossier bv. De oplossing zou zijn van een `this.store.query` te schrijven die `pieces` ophaalt voor een bepaalde case id en ook de page size expliciteert. Dit is hier echter niet nodig. Zie punt 2.
- Gezien je enkel maar een aantal nodig hebt, en niet de objecten zelf, kunnen we efficienter werken dan dit. Ongeacht van de page-size, bevat het resultaat van een `this.store.query` een `count`-metadata, die steeds een getal voor het totale aantal elementen zal bevatten. Zo kan je het totale aantal resultaten opvragen zonder dat de inhoud van de eigenlijke objecten meegestuurd moet worden. [Dit voorbeeld](https://github.com/kanselarij-vlaanderen/frontend-overlegcomite/blob/360566fd4915fe28b1507934196e4eaab1cfa53e/app/services/rightless-user-count.js#L19) gebruikt nog de oude `then`-promise-notatie, maar zou je voldoende op weg moeten helpen.